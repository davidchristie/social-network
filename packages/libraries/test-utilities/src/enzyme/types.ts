import { ReactWrapper, ShallowWrapper } from "enzyme";

export type Wrapper = ReactWrapper | ShallowWrapper;
