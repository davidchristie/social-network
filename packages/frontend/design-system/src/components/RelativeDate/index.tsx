import { distanceInWordsToNow } from "date-fns";
import React from "react";

interface Props {
  value: string;
}

export default class RelativeDate extends React.Component<Props> {
  public render () {
    const date = new Date(this.props.value);
    return distanceInWordsToNow(date);
  }
}
