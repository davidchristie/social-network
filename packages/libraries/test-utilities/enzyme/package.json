{
  "main": "../dist/enzyme/index.js",
  "name": "test-utilities/enzyme",
  "private": true,
  "version": "1.0.0"
}
