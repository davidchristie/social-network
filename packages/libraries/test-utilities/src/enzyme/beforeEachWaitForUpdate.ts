import { Wrapper } from "./types";

export default function beforeEachWaitForUpdate (getWrapper: () => Wrapper) {
  beforeEach(async () => {
    await new Promise(resolve => window.setTimeout(resolve, 0));
    getWrapper().update();
  });
}
