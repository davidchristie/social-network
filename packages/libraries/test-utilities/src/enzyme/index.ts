export { default as beforeEachWaitForUpdate } from "./beforeEachWaitForUpdate";
export { default as itContainsComponent } from "./itContainsComponent";
