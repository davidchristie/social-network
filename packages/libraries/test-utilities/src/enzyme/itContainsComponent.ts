import React from "react";
import { Wrapper } from "./types";

export default function itContainsComponent (
  name: string,
  getWrapper: () => Wrapper,
  Component: React.ComponentType
) {
  it(name, () => {
    expect(getWrapper().find(Component).exists()).toBe(true);
  });
}
