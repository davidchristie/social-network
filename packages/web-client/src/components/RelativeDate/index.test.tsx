import { shallow } from "enzyme";
import React from "react";

import RelativeDate from ".";

describe("RelativeDate component", () => {
  it("matches snapshot", () => {
    Date.now = () => 1539331728398;
    const wrapper = shallow(
      <RelativeDate value="2018-10-12T07:48:03.948Z" />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
