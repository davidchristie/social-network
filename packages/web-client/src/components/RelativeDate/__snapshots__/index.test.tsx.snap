// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RelativeDate component matches snapshot 1`] = `"21 minutes"`;
